### PR TITLE
Changing default hugepagesize request to 2MB

### PR DESCRIPTION
**Pull request by pkaminsk**
With fallback to default size on failure.

### CODE REVIEW

File: src/bf_sal/linux_usr/bf_sys_dma_hugepages.c [hunk 1]
```diff
@@ -181,11 +181,17 @@ static uint32_t conv_to_next_aligned(uint32_t val, uint32_t alignment) {
 static void *alloc_huge_pages(size_t size, unsigned int header_offset) {
   size_t actual_size;
   char *ptr;
-
   actual_size = ALIGN_TO_BF_PAGE_SIZE(size + header_offset);
   ptr = (char *)mmap(NULL, actual_size, PROT_READ | PROT_WRITE,
-                     MAP_SHARED | MAP_ANONYMOUS | MAP_POPULATE | MAP_HUGETLB,
+                     MAP_SHARED | MAP_ANONYMOUS | MAP_POPULATE | MAP_HUGETLB |
+                         (21 << MAP_HUGE_SHIFT),
```
Comment (slicking) [inline src/bf_sal/linux_usr/bf_sys_dma_hugepages.c:187]: MAP_HUGE_2MB may be more clear:
```suggestion
                         MAP_HUGE_2MB,
```

Comment (pkaminsk) [reply src/bf_sal/linux_usr/bf_sys_dma_hugepages.c:187]: I tried it but it doesn't compile on my system:
`/root/p4factory2/submodules/target-syslibs/src/bf_sal/linux_usr/bf_sys_dma_hugepages.c: In function â€˜alloc_huge_pagesâ€™:
/root/p4factory2/submodules/target-syslibs/src/bf_sal/linux_usr/bf_sys_dma_hugepages.c:187:26: error: â€˜MAP_HUGE_2MBâ€™ undeclared (first use in this function); did you mean â€˜MAP_HUGETLBâ€™?
  187 |                          MAP_HUGE_2MB,
      |                          ^~~~~~~~~~~~
      |                          MAP_HUGETLB
`

**Review by slicking — APPROVED** [review]

**Comment by saynb** [comment]
I am wondering if this will be a problem if the default is something other than 2MB and that's what the user wants to configure. Shouldn't we try and configure the default first and if that fails, then go and try the 2MB config @slicking @pkaminsk

**Comment by saynb** [comment]
But how did we come to 2 MB? Is it a number that is widely used ?

**Comment by slicking** [comment]
> But how did we come to 2 MB? Is it a number that is widely used ?

@saynb yes, the two common sizes are 2MB and 1G.

**Pull request merged**